openshift/template.yaml

```diff
@@ -0,0 +1,251 @@
+    resources:
+    - configmaps
+    - secrets
+    - serviceaccounts
+    verbs:
+    - list
+    - watch
+    - get
+  - apiGroups:
+    - triggers.tekton.dev
+    resources:
+    - '*'
+    verbs:
+    - '*'
+  - apiGroups:
+    - tekton.dev
+    resources:
+    - '*'
+    verbs:
+    - '*'
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: ClusterRole
+  metadata:
+    creationTimestamp: null
+    name: cad-eventlistener-clusterrole
+  rules:
+  - apiGroups:
+    - triggers.tekton.dev
+    resources:
+    - clusterinterceptors
+    verbs:
+    - list
+    - watch
+- Subjects:
+  - kind: ServiceAccount
+    name: cad-sa
+  apiVersion: rbac.authorization.k8s.io/v1
+  kind: RoleBinding
+  metadata:
+    name: cad-eventlistener-rolebinding
+  roleRef:
+    apiGroup: rbac.authorization.k8s.io
+    kind: Role
+    name: cad-eventlistener-role
+  subjects:
+  - kind: ServiceAccount
+    name: cad-sa
+- Subjects:
+  - kind: ServiceAccount
+    name: cad-sa
+  apiVersion: rbac.authorization.k8s.io/v1
+  kind: ClusterRoleBinding
+  metadata:
+    name: cad-eventlistener-clusterrolebinding
+  roleRef:
+    apiGroup: rbac.authorization.k8s.io
+    kind: ClusterRole
+    name: cad-eventlistener-clusterrole
+  subjects:
+  - kind: ServiceAccount
+    name: cad-sa
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    name: cad-aws-credentials
+  stringData:
+    AWS_ACCESS_KEY_ID: CHANGEME
+    AWS_DEFAULT_REGION: CHANGEME
+    AWS_SECRET_ACCESS_KEY: CHANGEME
+    AWS_SESSION_TOKEN: CHANGEME
+  type: Opaque
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    name: cad-ocm-client-secret
+  stringData:
+    CAD_OCM_CLIENT_ID: CHANGEME
+    CAD_OCM_CLIENT_SECRET: CHANGEME
+    CAD_OCM_URL: https://api.stage.openshift.com
+  type: Opaque
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    name: cad-pd-token
+  stringData:
+    token: CHANGEME
+  type: Opaque
+- apiVersion: tekton.dev/v1beta1
+  kind: Task
+  metadata:
+    name: cad-checks
+  spec:
+    params:
+    - description: Json string of the event data
+      name: payload
+      type: string
+    steps:
+    - args:
+      - |-
+        # save the payload to a file
+        folder=$(mktemp -d)
+        file=${folder}/payload.json
+        echo '$(params.payload)' > $file
+        # run the cadctl command
+        RESULT=$(cadctl cluster-missing --payload-path $file )
+      command:
+      - /bin/bash
+      - -c
+      env:
+      - name: CAD_PD
+        valueFrom:
+          secretKeyRef:
+            key: token
+            name: cad-pd-token
+      - name: CAD_ESCALATION_POLICY
+        value: XXXX
+      - name: CAD_SILENT_POLICY
+        value: XXXX
+      - name: CSS_JUMPROLE
+        value: XXXX
+      - name: SUPPORT_JUMPROLE
+        value: XXXX
+      envFrom:
+      - secretRef:
+          name: cad-aws-credentials
+      - secretRef:
+          name: cad-ocm-client-secret
+      image: quay.io/app-sre/cadctl@sha256:22d2a957d935e883f45addd35acc87450ce71ea2f94c33f2df70ff36b015486a
```

Comment:
this should be parameterized

Reply:
yes, but to which value?

Reply:
https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/docs/app-sre/continuous-delivery-in-app-interface.md#automatically-generated-parameters

Reply:
Ah yes, I forgot about this one. We want to parametrize this and very likely we have to change the names for the secrets as well

Reply:
I will keep this up as @ninabauer might change the secrets in her PR